crates/goose-cli/src/session/builder.rs

```diff
@@ -322,13 +321,28 @@ pub async fn build_session(session_config: SessionBuilderConfig) -> CliSession {
     let extensions_to_run: Vec<_> = if let Some(extensions) = session_config.extensions_override {
         agent.disable_router_for_recipe().await;
         extensions.into_iter().collect()
+    } else if session_config.resume {
+        if let Some(session_id) = session_id.as_ref() {
+            match SessionManager::get_session(session_id, false).await {
+                Ok(session_data) => {
+                    // Try to load saved extension configs directly
+                    if let Some(saved_state) =
+                        EnabledExtensionsState::from_extension_data(&session_data.extension_data)
+                    {
+                        // Use the saved configs as-is (no lookup needed!)
+                        saved_state.extensions
```

Comment:
What happens if I run a recipe with extensions A and B, then I later uninstall (not disable) extension B, and then resume that session? For consistency/repeatability you'd want B, but as a user I think I would expect Goose to never use B as long as I've uninstalled it.

Should it fail? Warn?